### PR TITLE
Docs: add an AlbumentationsX tutorial for custom DINO views

### LINKED ISSUE #2047: Tutorial proposal: customize and inspect DINO views with AlbumentationsX
Would you be open to a small standalone tutorial showing how to change DINO global and local view policies and inspect a selected realization without copying Lightly's full `DINOTransform`?

## Motivation

Issue #1814 documents a concrete limitation: changing the `RandomResizedCrop` ratio currently requires reimplementing most of the DINO transform. The maintainer discussion there also considers a more composable transform abstraction, while noting that the refactor is not currently prioritized.

A focused tutorial could give users a documented composition pattern with the current Lightly API. Lightly would continue to define the DINO view roles, dataset contract, model, and training workflow. AlbumentationsX would define the image pipeline for each global or local view.

The user could then change crop ratio, scale, or photometric operations directly, save each configured policy with `A.to_dict()`, and reproduce one diagnostic view with `run_with_trace(..., invocation_seed=...)` to inspect which transforms ran and which parameters they sampled.

## Proposed tutorial

The tutorial would:

1. Define separate AlbumentationsX pipelines for DINO's two global views and its local views, including the intended blur and solarization differences.
2. Adapt each pipeline from a PIL image to the tensor returned by `A.ToTensorV2()`.
3. Pass those callables to Lightly's `MultiViewTransform` and use the normal Lightly training contract.
4. Serialize the configured view policies with `A.to_dict()`.
5. Run one selected image through `run_with_trace()` with explicit per-view `invocation_seed` values, so the same diagnostic views and sampled parameters can be regenerated.

A small sketch of the composition boundary is:

```python
import numpy as np
import albumentations as A

from lightly.transforms import MultiViewTransform


class AXView:
    def __init__(self, pipeline: A.Compose) -> None:
        self.pipeline = pipeline

    def __call__(self, image):
        return self.pipeline(image=np.asarray(image))["image"]


def make_view(size: int, scale: tuple[float, float]) -> A.Compose:
    return A.Compose(
        [
            A.RandomResizedCrop(
                size=(size, size),
                scale=scale,
                ratio=(0.7, 1.4),
                p=1.0,
            ),
            A.HorizontalFlip(p=0.5),
            A.Normalize(),
            A.ToTensorV2(),
        ]
    )


global_view_0 = make_view(size=224, scale=(0.4, 1.0))
global_view_1 = make_view(size=224, scale=(0.4, 1.0))
local_view = make_view(size=96, scale=(0.05, 0.4))

transform = MultiViewTransform(
    transforms=[AXView(global_view_0), AXView(global_view_1)]
    + [AXView(local_view)] * 6
)
```

The full tutorial would use the DINO-specific photometric policies; the sketch keeps only the code needed to show the Lightly–AX boundary.

## Scope and placement

This would be a documentation tutorial. It would not change `DINOTransform`, `MultiViewTransform`, or Lightly's runtime dependencies. The existing legacy Albumentations material in `structure_your_input.rst` would remain unchanged.

PR #2037 is currently exploring new view contracts for LightlySSL 2.0. Which contract should the tutorial target: the current API or the proposed 2.0 view API?

If the tutorial fits the project, I can prepare a documentation-only PR after your guidance on placement and target API.

## Optional dependency boundary

The tutorial would be explicitly optional and require Python 3.10 or newer. The public AlbumentationsX package is AGPL-3.0-only. Importing it also requires PyTorch; PyTorch is intentionally absent from its package metadata because the user must select the CPU, CUDA, or MPS build required by the application. The tutorial would document that installation boundary. AlbumentationsX would remain outside Lightly's runtime dependencies.

### PR DESCRIPTION
**Pull request by ternaus**
Closes #2047

## Description

- [ ] My change is breaking

Adds a Sphinx Gallery tutorial for customizing DINO crop and photometric policies with optional AlbumentationsX. It uses Lightly's current `MultiViewTransform` to produce two global and six local tensors, then shows dataset integration, policy serialization, and seeded tracing of sampled transforms. A restored policy and the same invocation seed reproduce the diagnostic view exactly.

AlbumentationsX stays outside Lightly's runtime dependencies. The installation instructions include the Matplotlib extra used by the tutorial and document Python 3.10+, the user-selected PyTorch build, and AlbumentationsX's AGPL-3.0-only license.

## Tests

- [x] My change is covered by existing tests.
- [ ] My change needs new tests.
- [ ] I have added/adapted the tests accordingly.
- [x] I have manually tested the change.

Executed the complete tutorial with `albumentationsx==2.4.3`; verified tensor and sampled-parameter reproduction for all three restored policies, fresh training samples, and an eight-view `LightlyDataset` batch. The focused transform tests passed (3 tests), along with full-tree Ruff checks, all hooks on the PR files, and Sphinx documentation and tutorial execution.

Repository-wide pre-commit still reports the existing codespell finding in unchanged `lightly/loss/frossl_loss.py:90` (`fro`). The docs build also has existing cross-reference warnings outside this tutorial.

## Documentation

- [x] I have added docstrings to all public functions/methods.
- [x] My change requires a change to the documentation (`.rst` files).
- [x] I have updated the documentation accordingly.
- [x] The autodocs update the documentation accordingly.

The generated tutorial and PDF preview were inspected visually. OpenCV and torchvision can differ pixel-wise; the tutorial recommends evaluating policy changes on the intended downstream task.

Adds a standalone Sphinx Gallery tutorial for customizing DINO global and local views with optional AlbumentationsX. The tutorial uses `MultiViewTransform`, supports configurable crop and photometric policies, serializes policies with `A.to_dict()`, and reproduces diagnostic views with `run_with_trace()`. Adds the tutorial to the documentation index without changing runtime dependencies or existing documentation.

### CODE REVIEW

**Comment by gabrielfruet** [comment]
Hey, is this ready to be reviewed?

**Comment by ternaus** [comment]
Yes, @gabrielfruet, this is ready for review now. I added the missing seeded diagnostics and verified the complete tutorial, policy reproduction, and Lightly dataset integration.

The focused tests and docs build pass. The GitHub Actions runs still need maintainer approval.
